lib/spack/spack/operating_systems/linux_distro.py

```diff
@@ -2,11 +2,22 @@
 # Spack Project Developers. See the top-level COPYRIGHT file for details.
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import platform as py_platform
 import re
 
+from spack.version import Version
+
 from ._operating_system import OperatingSystem
 
 
+def kernel_version():
+    """Return the kernel version as a Version object"""
+    # Strip '+' characters just in case we're running a
+    # version built from git/etc
+    clean_version = re.sub(r'\+', r'', py_platform.release())
+    return Version(clean_version)
```

Comment:
I'm certainly willing to move the code around to where we think it makes sense - but for clarity on what we're trying to obtain (and why kernel_version is distinct from an analogue to macos_version, or linux_version), this should highlight the differences:
```
coltrane (/home/aweits):spack-python
Spack version 0.17.0
Python 3.6.8, Linux x86_64
>>> import external.distro
>>> import platform
>>> external.distro.id()
'centos'
>>> external.distro.version()
'7'
>>> platform.release()
'5.10.84+'
```

Reply:
Thanks for the clarification, my suggestion definitely won't work. I think it would make sense to update the docstring to mention that the kernel version is distinct from the OS/Distribution version (and perhaps use the values you printed here to demonstrate the difference).